Bottom: d3f75c0dddd9eae4972c02cec54f405e6068e150
Top:    a4fffdce130e2da49412f6d79d91214dd1871d36
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2016-01-06 23:12:40 +0000

Try a new, different sized VBO buffer for each upload.


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 32b6dc5..66586f1 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -153,10 +153,6 @@ hidgl_init_triangle_array (hidgl_instance *hidgl)
    */
   priv->buffer.use_map = false;
 
-  /* If using VBOs (but not mapping), we only need to this once */
-  if (priv->buffer.use_vbo && !priv->buffer.use_map)
-    glBufferData (GL_ARRAY_BUFFER, BUFFER_SIZE, NULL, GL_STREAM_DRAW);
-
   priv->buffer.triangle_array = NULL;
   hidgl_reset_triangle_array (hidgl);
 }
@@ -198,10 +194,10 @@ hidgl_flush_triangles (hidgl_instance *hidgl)
       glUnmapBuffer (GL_ARRAY_BUFFER);
       priv->buffer.triangle_array = NULL;
     } else {
-      /* NB: We only upload the portion of the buffer we've used */
-      glBufferSubData (GL_ARRAY_BUFFER, 0,
-                       BUFFER_STRIDE * priv->buffer.vertex_count,
-                       priv->buffer.triangle_array);
+      glBufferData (GL_ARRAY_BUFFER,
+                    BUFFER_STRIDE * priv->buffer.vertex_count,
+                    priv->buffer.triangle_array,
+                    GL_STREAM_DRAW);
     }
   } else {
     data_pointer = priv->buffer.triangle_array;
